mycroft/util/settings_gui_generator.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2020 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import yaml
+
+
+class SettingsGuiGenerator:
+    """Skill Settings Generator For GUI. """
+
+    def __init__(self):
+        """ Create a SettingList Object """
+        self.settings_list = []
+
+    def populate(self, skill_id, settings_file, settings_dict, file_type):
```

Comment:
Could the `file_type` be dropped and `os.path.splitext(settings_file)` or `pathlib.Path(settings_file).suffix` be used to detect the file_type?

Reply:
replaced with pathlib.Path(settings_file).suffix